integrationtests/src/test/java/io/kroxylicious/test/kafkacluster/extension/KafkaClusterExtensionTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.test.kafkacluster.extension;
+
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.DescribeClusterResult;
+import org.apache.kafka.common.errors.SaslAuthenticationException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.kroxylicious.test.kafkacluster.ContainerBasedKafkaCluster;
+import io.kroxylicious.test.kafkacluster.KafkaCluster;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@ExtendWith(KafkaClusterExtension.class)
```

Comment:
This is possibly a use case for nested test cases. So that the unit test for the extension doesn't depend on the extension itself. I'd almost make the argument that this is an "integration" test.